common/utils/src/main/scala/org/apache/spark/util/DayTimeIntervalUtils.java

```diff
@@ -0,0 +1,134 @@
+      String from = fieldToString(startField).toUpperCase();
+      String to = fieldToString(endField).toUpperCase();
+      String prefix = "INTERVAL '";
+      String postfix = startField == endField ? "' " + from : "' " + from + " TO " + to;
+      if (micros < 0) {
+        if (micros == Long.MIN_VALUE) {
+          // Especial handling of minimum `Long` value because negate op overflows `Long`.
+          // seconds = 106751991 * (24 * 60 * 60) + 4 * 60 * 60 + 54 = 9223372036854
+          // microseconds = -9223372036854000000L-775808 == Long.MinValue
+          String baseStr = "-106751991 04:00:54.775808000";
+          String firstStr = "-" + (startField == DAY ? Long.toString(MAX_DAY) :
+                  (startField == HOUR ? Long.toString(MAX_HOUR) :
+                          (startField == MINUTE ? Long.toString(MAX_MINUTE) :
+                                  Long.toString(MAX_SECOND) + ".775808")));
+          if (startField == endField) {
+            return prefix + firstStr + postfix;
+          } else {
+            int substrStart = startField == DAY ? 10 : (startField == HOUR ? 13 : 16);
+            int substrEnd = endField == HOUR ? 13 : (endField == MINUTE ? 16 : 26);
+            return prefix + firstStr + baseStr.substring(substrStart, substrEnd) + postfix;
+          }
+        } else {
+          sign = "-";
+          rest = -rest;
+        }
+      }
+      StringBuilder formatBuilder = new StringBuilder(sign);
+      ArrayList<Long> formatArgs = new ArrayList<>();
+      if (startField == DAY) {
+        formatBuilder.append(rest / MICROS_PER_DAY);
+        rest %= MICROS_PER_DAY;
+      } else if (startField == HOUR) {
+        formatBuilder.append("%02d");
+        formatArgs.add(rest / MICROS_PER_HOUR);
+        rest %= MICROS_PER_HOUR;
+      } else if (startField == MINUTE) {
+        formatBuilder.append("%02d");
+        formatArgs.add(rest / MICROS_PER_MINUTE);
+        rest %= MICROS_PER_MINUTE;
+      } else if (startField == SECOND) {
+        String leadZero = rest < 10 * MICROS_PER_SECOND ? "0" : "";
+        formatBuilder.append(leadZero + BigDecimal.valueOf(rest, 6)
+                .stripTrailingZeros().toPlainString());
+      }
+
+      if (startField < HOUR && HOUR <= endField) {
+        formatBuilder.append(" %02d");
+        formatArgs.add(rest / MICROS_PER_HOUR);
+        rest %= MICROS_PER_HOUR;
+      }
+      if (startField < MINUTE && MINUTE <= endField) {
+        formatBuilder.append(":%02d");
+        formatArgs.add(rest / MICROS_PER_MINUTE);
+        rest %= MICROS_PER_MINUTE;
+      }
+      if (startField < SECOND && SECOND <= endField) {
+        String leadZero = rest < 10 * MICROS_PER_SECOND ? "0" : "";
+        formatBuilder.append(":" + leadZero + BigDecimal.valueOf(rest, 6)
+                .stripTrailingZeros().toPlainString());
+      }
+      return prefix + String.format(formatBuilder.toString(), formatArgs.toArray()) + postfix;
+    } catch (SparkException e) {
```

Comment:
The `try-catch` here seems a bit redundant. Why do we need to catch the `SparkException` only to rethrow it?